frontend/src/features/admin-form/responses/common/utils/mrfSubmissionView.ts

```diff
@@ -69,3 +69,35 @@ export const getStatusFromWorkflowStatus = (
     }
   }
 }
+
+export type getWorkflowStatusFromFormResponseProps = {
+  index: number
+  workflow: FormWorkflowDto
+  submittedSteps: SubmittedStep[]
+}
+
+/** Gets a step's workflow status from workflow steps and MRF submmission steps */
+export const getWorkflowStatusFromFormResponse = ({
+  index,
+  workflow,
+  submittedSteps,
+}: getWorkflowStatusFromFormResponseProps): WorkflowStatus => {
+  const submittedStep = submittedSteps[index]
+
+  // Steps that haven't been submitted must be PENDING
+  if (index < submittedSteps.length) {
+    return WorkflowStatus.PENDING
+  }
```

Comment:
So sorry, I messed up the condition here. [The pending and completed steps got inverted](https://staging-alt2.form.gov.sg/686cb973620e0fbd924cdf66/status/686cba95620e0fbd924ce16d).
<img width="1624" alt="image" src="https://github.com/user-attachments/assets/8576162c-974c-4321-a92e-98029aa9b71e" />

```suggestion
  if (index >= submittedSteps.length) {
    return WorkflowStatus.PENDING
  }
```

Reply:
this should invert the [previous conditional](https://github.com/opengovsg/FormSG/pull/8529/commits/81f5d36d3f4b19b5625e4faece913069c23ca1d7#diff-82da138f99b5b6b97af343f2860c82579df8cc0cd5d0f7fb208f0e5058cfcbf7R86) `index < submittedSteps.length` correctly